py/desispec/test/test_io.py

```diff
@@ -1202,11 +1202,25 @@ def test_get_tempfilename(self):
         tempfile = get_tempfilename(filename)
         self.assertNotEqual(filename, tempfile)
         self.assertTrue(tempfile.endswith('.fits'))
+        self.assertTrue('/a/b/c' in tempfile)
+
+        filename = '/a/b/blat.fits.gz'
+        tempfile = get_tempfilename(filename)
+        self.assertNotEqual(filename, tempfile)
+        self.assertTrue(tempfile.endswith('.fits.gz'))
+        self.assertTrue('/a/b/blat' in tempfile)
+
+        filename = '/a/b/blat.fits.fz'
+        tempfile = get_tempfilename(filename)
+        self.assertNotEqual(filename, tempfile)
+        self.assertTrue(tempfile.endswith('.fits.fz'))
+        self.assertTrue('/a/b/blat' in tempfile)
```

Comment:
Your updated `get_tempfilename` conveniently "does the right thing" even if there is a single `.gz` extension, i.e. `blat.gz` -> `blat_tmp14074.gz` and `blat.fits.gz` -> `blat_tmp14074.fits.gz`.  Let's add a test to explicitly check that since we don't want to accidentally break it in the future and some other implementation might crash with a single extension `.gz` .

Reply:
This should be resolved in 638e35f